app/screens/EmissionItem/EmissionItemScreen.tsx

```diff
@@ -35,60 +35,75 @@ const EmissionItemScreen = ({ language = "" }: LocalizationContextProps) => {
   // eslint-disable-next-line @typescript-eslint/no-empty-function
   const onPress = () => {};
 
-  /* Avoid crash right after an emission is deleted */
-  if (!emission) {
-    navigator.goBack();
-    return null;
-  }
-
-  const { creationDate = "", emissionModelType = "", name = "" } = emission;
+  const { creationDate, emissionModelType, name } = emission || {
+    creationDate: new Date(),
+    emissionModelType: "",
+    name: "",
+  };
 
   const date = moment(creationDate, "YYYY-MM-DDTHH:mm:ss.sssZ");
   const day = date.locale(language).format("dddd");
   const monthAndYear = date.locale(language).format("Do MMMM YYYY");
-  const co2Emission = calculation.getC02ValueFromEmission(emission);
-  const deleteEmission = () =>
+  const co2Emission = calculation.getC02ValueFromEmission(emission || {});
+  const deleteEmission = () => {
     dispatch(emissions.actions.deleteEmissionById(emission.id));
+    navigator.goBack();
```

Comment:
Delete this line otherwise we have a double goBack because of the useEffect

Reply:
Done!